students/daniel_grubbs/lesson09/mailroom_oo.py

```diff
@@ -0,0 +1,158 @@
+#!/usr/bin/env python3
+
+import sys
+
+# Define and global variables and constants
+donors = {'Jimmy Nguyen': [100, 1350, 55],
+          'Steve Smith': [213, 550, 435],
+          'Julia Norton': [1500, 1500, 1500],
+          'Ed Johnson': [150],
+          'Elizabeth McBath': [10000, 1200]
+          }
+
+
+class Donor(object):
+    """
+    Hold all the donor objects, amd also methods to add a new donor,
+    search for a given donor, etc...
+    """
+
+    # def __init__(self):
+    #     """Initializes the Donor object with a given name and list of donations."""
+    #     self.name = name
+
+    def show_donor_names(self):
+        """Shows a listing of the current donors. For use when listing out current donors."""
+        donor_names = []
+        for k in donors.keys():
+            donor_names.append(k)
+        return donor_names
+
+    def donor_lookup(full_name):
+        """Search the dictionary keys and look for a donor by their full name."""
+        donor = full_name.strip()
+        for k in donors.keys():
+            if donor in donors:
+                return donor
+            else:
+                return None
+
+    def add_new_donor(full_name):
+        """Adds a new donor to the donor dictionary."""
+        full_name = full_name.strip()
+        new_donor = (full_name, [])
+        donors[full_name] = new_donor
+        return new_donor
+
+    def create_donor_report(self):
+        """Create a report of the donors and donation amounts."""
+        donations = []
+
+        print("{:26s} | {:13s} | {:9s} | {:13s}".format("Donor name", "Total Donation", "Number of Gifts",
+                                                        "Average Gifts"))
+        print("-" * 80)
+
+        for donor, gift in donors.items():
+            total_given = sum(gift)
+            number_gifts = len(gift)
+            average_gift = total_given / number_gifts
+            donations.append((donor, total_given, number_gifts, average_gift))
+
+        for amount in donations:
+            print("{:26s} | {:14.2f} | {:15d} | {:13.2f}".format(*amount))
+        print()
+
+    def letter_template(self, donor):
+        """Template for writing a letter to a donor, thanking them for their donation."""
+        return """Dear {},\nThank you for your very kind donation of {:.2f}.\n\nIt will be put to very good use.\n\n \t\tSincerely,\n\t\t\t-The Team""".format(
+            donor, donors[donor][-1])
+
+    def send_letter_file(self):
```

Comment:
These two functions `letter_template` and `send_letter_file` should not be in this class as they aren't mutating or changing the donor class in some way.  That isn't to say that its required to ONLY have things that mutate or change that class, but it just doesn't make sense to have them in here.